### DISCUSSION TITLE
Way to run same plugin twice in the request context ?

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#8686] [Help] [answered]
Hi,

Is there any way to run same plugin twice in the request context with different priority ( and may be different config) ? Want to run the same plugin before and after auth

Thanks

**Comment by hbagdi** [comment, marked as answer]
Can you elaborate on the use-case like which plugin and why?
We are working on solving this problem within the product itself.
And to answer your question, this is not possible today.
One way to do this is to duplicate the code of the plugin and install the duplicated code with a different name as a custom plugin, with appropriate priority (execution priorities are hardcoded in the plugin code) and then use that plugin in your configuration.

**Reply by shettyh** [reply]
Thanks for the response, yes the current approach is same as you mentioned by duplicating code.

>We are working on solving this problem within the product itself.

Great to hear that. Any timelines ?

> Can you elaborate on the use-case like which plugin and why?

Plugin is custom, that takes decision to accept or reject request based on few parameters. Some of these parameters are directly available in the request but some are derived during the AuthN/AuthZ. Currently this custome plugin runs after the auth plugins, but we wanted to optimise this by running this plugin before and after auth. This will help in reducing the load on Kong in case we could reject lot of requests just by using the request attributes.
